aws/keys/ec.go

```diff
@@ -0,0 +1,41 @@
+package keys
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"io"
+
+	"github.com/aws/aws-sdk-go/service/kms"
+)
+
+// AwsEcPublicKey represents the public part of an ECDSA key.
+type AwsEcPublicKey struct {
+	*ecdsa.PublicKey
+	ARN              string
+	Region           string
+	AwsPublicKeyInfo kms.GetPublicKeyOutput
+}
+
+// AwsEcPrivateKey represents an ECC key
+type AwsEcPrivateKey struct {
+	AwsEcPublicKey // public part.
+}
+
+// Public returns the public key corresponding to priv.
+func (priv AwsEcPrivateKey) Public() crypto.PublicKey {
```

Comment:
Not using a pointer with struct in the receiver declaration makes copies of the struct every time you call it.